### PR TITLE
refactor: Update AWS Glue Databrew StartJobRun step to use integration pattern input

### PR DESCRIPTION
**Pull request by ca-nguyen**
### Description

Update AWS Glue Databrew service integration to use `integration_pattern` input instead or `wait_for_completion` flag.

Fixes #(issue) (N/A)

### Why is the change necessary?
This change is necessary for consistency with the new service integration implementation pattern introduced in [commit](https://github.com/aws/aws-step-functions-data-science-sdk-python/commit/f9e63b312f543954c663e305fd5eedd9b2cdd360) (Add support for Nested Step Functions) that uses the `integration_pattern` arg in the step constructor to build the resource.

Support for AWS Glue Databrew service integration was added in this [commit](https://github.com/aws/aws-step-functions-data-science-sdk-python/commit/ac652ab84ff74284ce996097c7c857e642043a3b), but not **released yet**.
A later [commit](https://github.com/aws/aws-step-functions-data-science-sdk-python/commit/f9e63b312f543954c663e305fd5eedd9b2cdd360) (Add support for Nested Step Functions) introduced a new implementation pattern using the `IntegrationPattern` enum as input to construct the step instead of the `wait_for_completion` flag. (See [PR](https://github.com/aws/aws-step-functions-data-science-sdk-python/pull/166) for more detail on rationale behind the implementation).

### Solution

 Replace the `wait_for_completion` flag with `integration_pattern` arg in [StartJobRun](https://docs.aws.amazon.com/step-functions/latest/dg/connect-databrew.html) step construction.
 
 The `IntegrationPattern` is used to build the `Resource` arn as follow:
| IntegrationPattern  | Resource                                                                                 | Doc
|:------------------------|:----------------------------------------------------------|:---|
| WaitForCompletion    | "arn:aws:states:::states:databrew:startJobRun.sync"                   | [Run A job](https://docs.aws.amazon.com/step-functions/latest/dg/connect-to-resource.html#connect-sync)|
| CallAndContinue                    | "arn:aws:states:::states:databrew:startJobRun"                                | [Request Response](https://docs.aws.amazon.com/step-functions/latest/dg/connect-to-resource.html#connect-default)|

See [Service Integration Patterns](https://docs.aws.amazon.com/step-functions/latest/dg/connect-to-resource.html) for more details
 
 Normally, replacing a constructor argument would be a breaking change, but since we have not released support for AWS Glue Databrew service integration yet, it is acceptable to do so. After next release, it making such changes will be considered as not being backward compatible.

### Testing

- Updated unit tests for all resources
- No integ tests added as external resources were required for testing (Databrew Job)
- Manual test done for all resources
#### Manual Tests
Create a `Databrew` job for manual tests (<databrew_job_name>). 
For each test, create a workflow and execute as follow:
```
workflow = Workflow(
    name=<state_machine_name>,
    definition=<step>,
    role=<workflow_execution_role>
)

workflow.create()
workflow.execute()
```
   - Call And Continue: 
```
# Create a workflow with the following steps & execute
step_call_and_continue = GlueDataBrewStartJobRunStep(
        'Start Glue DataBrew Job Run - CallAndContinue',
        integration_pattern=IntegrationPattern.CallAndContinue, parameters={
            "Name": <databrew_job_name>
        }
)

step_default = GlueDataBrewStartJobRunStep('Start Glue DataBrew Job Run - Default', parameters={
        "Name":  <databrew_job_name>
    }
)

# Validate that the workflows end after starting the  <databrew_job_name> job
```
   - Wait For Completion
```
# Create a workflow with the following step & execute
step_wait_for_completion = GlueDataBrewStartJobRunStep(
        'Start Glue DataBrew Job Run - WaitForCompletion',
        integration_pattern=IntegrationPattern.WaitForCompletion, parameters={
            "Name":  <databrew_job_name>
        }
)
# Validate that the workflow only ends after the  <databrew_job_name> job completes
```

----

### Pull Request Checklist

Please check all boxes (including N/A items)

#### Testing

- [X] Unit tests added
- [X] Integration test added - **N/A**
- [X] Manual testing - why was it necessary? could it be automated?  **No integ tests added as external resources were required for testing** 

#### Documentation

- [X] __docs__: All relevant [docs](https://github.com/aws/aws-step-functions-data-science-sdk-python/tree/main/doc) updated
- [X] __docstrings__: All public APIs documented

### Title and description

- [X] __Change type__: Title is prefixed with change type: and follows [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [X] __References__: Indicate issues fixed via: `Fixes #xxx` - **N/A**

----

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license.

### CODE REVIEW

**Review by shivlaks — APPROVED** [review]

File: tests/unit/test_service_steps.py [hunk 1]
```diff
@@ -692,10 +692,30 @@ def test_databrew_start_job_run_step_creation_sync():
...
-def test_databrew_start_job_run_step_creation():
-    step = GlueDataBrewStartJobRunStep('Start Glue DataBrew Job Run', wait_for_completion=False, parameters={
-        "Name": "MyWorkflowJobRun"
-    })
+def test_databrew_start_job_run_step_creation_wait_for_completion():
+    step = GlueDataBrewStartJobRunStep(
+        'Start Glue DataBrew Job Run - WaitForCompletion', integration_pattern=IntegrationPattern.WaitForCompletion,
+        parameters={
+            "Name": "MyWorkflowJobRun"
+        })
```
Comment (shivlaks) [inline tests/unit/test_service_steps.py:700]: I think there should be a test for verifying the invalid (unsupported) service integration pattern as well. This was not a case that was previously possible but needs test coverage now.

**Review by shivlaks — CHANGES_REQUESTED** [review]

File: tests/unit/test_service_steps.py (hunk 1 above)
Comment (ca-nguyen) [reply tests/unit/test_service_steps.py:700]: Agreed - we should cover this case as well! 
Changes will be added in the next commit

Comment (ca-nguyen) [reply tests/unit/test_service_steps.py:700]: Added unit test to cover unsupported integration pattern in last commit

File: tests/unit/test_service_steps.py [hunk 2]
```diff
@@ -676,8 +676,8 @@ def test_emr_modify_instance_group_by_name_step_creation():
 
 
 @patch.object(boto3.session.Session, 'region_name', 'us-east-1')
-def test_databrew_start_job_run_step_creation_sync():
-    step = GlueDataBrewStartJobRunStep('Start Glue DataBrew Job Run - Sync', parameters={
+def test_databrew_start_job_run_step_creation_default():
+    step = GlueDataBrewStartJobRunStep('Start Glue DataBrew Job Run - Default', parameters={
```
Comment (shivlaks) [inline tests/unit/test_service_steps.py:680]:
```suggestion
    step = GlueDataBrewStartJobRunStep('Start Glue DataBrew Job Run - default', parameters={
```

nit: address before pushing if you agree

File: tests/unit/test_service_steps.py [hunk 3]
```diff
@@ -707,6 +727,13 @@ def test_databrew_start_job_run_step_creation():
     }
 
 
+@patch.object(boto3.session.Session, 'region_name', 'us-east-1')
+def test_databrew_start_job_run_step_creation_wait_for_task_token_raises_error():
+    with pytest.raises(ValueError):
+        GlueDataBrewStartJobRunStep('Start Glue DataBrew Job Run - WaitForTaskToken',
+                                    integration_pattern=IntegrationPattern.WaitForTaskToken)
```
Comment (shivlaks) [inline tests/unit/test_service_steps.py:734]: nit: can we assert the message as well? I think `pytest.raises` accepts a `match=<message>` - [documentation](https://docs.pytest.org/en/stable/reference.html?highlight=raises#pytest.raises)

Comment (ca-nguyen) [reply tests/unit/test_service_steps.py]: Yes - adding a validation on the error message will ensure no breaking changes are introduced

Added error message assert in the last commit

**Review by shivlaks — APPROVED** [review]

**Review by evscott — APPROVED** [review]

File: src/stepfunctions/steps/service.py [hunk 4]
```diff
@@ -563,23 +567,18 @@ def __init__(self, state_id, wait_for_completion=True, **kwargs):
...
-            """
+        supported_integ_patterns = [IntegrationPattern.WaitForCompletion, IntegrationPattern.CallAndContinue]
 
-            kwargs[Field.Resource.value] = get_service_integration_arn(GLUE_DATABREW_SERVICE_NAME,
-                                                                       GlueDataBrewApi.StartJobRun)
+        is_integration_pattern_valid(integration_pattern, supported_integ_patterns)
+        kwargs[Field.Resource.value] = get_service_integration_arn(GLUE_DATABREW_SERVICE_NAME,
+                                                                   GlueDataBrewApi.StartJobRun,
+                                                                   integration_pattern)
+        """
```
Comment (jormello) [inline src/stepfunctions/steps/service.py:577]: Nit (non-blocking): would be good to have the comment precede the Field.Resource assignment.

**Review by jormello — APPROVED** [review]

**Review by shivlaks — APPROVED** [review]

**Pull request merged**